### ISSUE TITLE
passwordRealm fails after migration to null safety

### ISSUE THREAD

**Issue by nickeliot** [mishatron/flutter_auth0#5]
We are trying to transition our application to null safe status, but are having issues with the Auth0 package.
When we execute the "passwordRealm" method, an exception is thrown.

```
    Auth0User passwordResponse = await auth0.passwordRealm({
      'username': username,
      'password': password,
      'realm': 'Username-Password-Authentication',
    });
```

`type 'Null' is not a subtype of type 'String'`

The message implies that a "string" value is not properly assigned, but we are failing to figure out what it is.
Any help or insight would be appreciated.

We had everything working before we migrated to null safety.

**Comment by mishatron** [comment]
Thanks, i will look at it today. Do you know at what place we have an issue?

**Comment by nickeliot** [comment]
It appears that the members of Auth0User are non-nullable. 
This is problematic because the Auth0 response for ...
```
Auth0User passwordResponse = await auth0.passwordRealm({
    'username': username,
    'password': password,
    'realm': 'Username-Password-Authentication',
});
```
does not return a _"refreshToken"_.

Similarly ...
```
Auth0User accessResponse = await auth0.passwordRealm({
      'username': username,
      'password': password,
      'realm': 'Username-Password-Authentication',
      'audience': 'https://AcmeLabs/Api',
      'scope': 'offline_access'
    });
```
does not return an _"idToken"_.

Other combinations 'audience' and 'scope' may generate issues as well.  We did not test further.

### Workaround

Expand 'scope' to force return of missing data.
For first case set 'scope' to 'offline_access'.
```
Auth0User passwordResponse = await auth0.passwordRealm({
    'username': username,
    'password': password,
    'realm': 'Username-Password-Authentication',
    'scope': 'offline_access'
});
```
In second case, add 'openid' to 'scope'.
```
Auth0User accessResponse = await auth0.passwordRealm({
      'username': username,
      'password': password,
      'realm': 'Username-Password-Authentication',
      'audience': 'https://AcmeLabs/Api',
      'scope': 'offline_access openid'
    });
```

**Comment by mishatron** [comment]
I released version 2.0.2 where all fileds of the Auth0User are nullable, so it should not cause any problems. I close issue for now
